src/ic-cdk-bindgen/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+use candid::{bindings::rust, pretty_check_file, Principal};
+use std::env;
+use std::fs;
+use std::io::Write;
+use std::path::PathBuf;
+
+#[derive(Clone)]
+pub struct Config {
+    pub canister_name: String,
+    pub candid_path: PathBuf,
+    pub skip_existing_files: bool,
+    pub binding: rust::Config,
+}
+impl Config {
+    pub fn new(canister_name: &str) -> Self {
+        let candid_path_var_name = format!("CANISTER_CANDID_PATH_{}", canister_name);
+        let candid_path =
+            PathBuf::from(env::var(candid_path_var_name).expect("Cannot find candid path"));
+        let canister_id_var_name = format!("CANISTER_ID_{}", canister_name);
+        let canister_id =
+            Principal::from_text(env::var(canister_id_var_name).expect("Cannot find canister id"))
+                .unwrap();
+        Config {
+            canister_name: canister_name.to_string(),
+            candid_path,
+            skip_existing_files: false,
+            binding: rust::Config {
+                // User will depend on candid crate directly
+                candid_crate: "candid".to_string(),
+                type_attributes: "".to_string(),
+                canister_id: Some(canister_id),
+                service_name: canister_name.to_string(),
+                target: rust::Target::CanisterCall,
+            },
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct Builder {
+    configs: Vec<Config>,
+}
+
+impl Builder {
+    pub fn new() -> Self {
+        Builder {
+            configs: Vec::new(),
+        }
+    }
+    pub fn add(&mut self, config: Config) -> &mut Self {
+        self.configs.push(config);
+        self
+    }
+    pub fn build(self, out_path: Option<PathBuf>) {
+        let out_path = out_path.unwrap_or_else(|| {
+            let manifest_dir =
+                PathBuf::from(env::var("CARGO_MANIFEST_DIR").expect("Cannot find manifest dir"));
+            // TODO dfx provide out_path
```

Comment:
Would you please file a ticket for the follow-up work and link to it here?

Reply:
I will remove this "TODO" comment.
The current implementation is sufficient.